### PR TITLE
make linode_domain_record.name computed for non-SRV records

### LINKED ISSUE #87: linode_domain_record always updates SRV name and target values
SRV domain records are requiring updates on every apply, even when there are no config changes.

### Terraform Version
```
$ terraform -v
Terraform v0.12.20
+ provider.linode v1.9.1
```

### Affected Resource(s)
- linode\_domain\_record

### Terraform Configuration Files
```hcl
terraform {
  required_version = ">= 0.12"
}

provider "linode" {
  version = "~> 1.8"
}

variable "domain"           { default = "CHANGEME.4zbug.stj.io" }
variable "domain_soa_email" { default = "webmaster@example.com" }

resource "linode_domain" "root" {
  type        = "master"
  domain      = var.domain
  soa_email   = var.domain_soa_email
}

resource "linode_domain_record" "a-oli" {
  domain_id   = linode_domain.root.id
  name        = "oli"
  record_type = "A"
  target      = "127.0.0.1"
}

resource "linode_domain_record" "srv-oli" {
  domain_id   = linode_domain.root.id
  name        = "oli"
  record_type = "SRV"
  target      = "oli-backend-1"
  service     = "oli"
  protocol    = "tcp"
  port        = 1001
  priority    = 10
  weight      = 0

  lifecycle {
    ignore_changes = [
      priority,
      weight,
    ]
  }
}

```

### Debug Output
CLI output for first and second apply.
https://gist.github.com/stvnjacobs/0168c72ff1f68ad4ea19868e92f3e465

The trace log output for the first and second apply.
https://gist.github.com/stvnjacobs/267a4155fa1f44ed3e3ba3ca5cd35d86#file-01-apply-log
https://gist.github.com/stvnjacobs/267a4155fa1f44ed3e3ba3ca5cd35d86#file-02-apply-log

### Expected Behavior
There should be no updates required for second apply.

### Actual Behavior
Updates to SRV name and target records are required for second apply.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Set `domain` variable
2. `terraform apply`
3. `terraform apply`

### PR DESCRIPTION
**Pull request by Charliekenney23**
Closes #87

This meant marking the field as `Computed` and `Optional`. There is a runtime check for non-SRV records to still require `name`. A validation step was also added to ensure `.target` for SRV record types are the associated domain or a FQDN related to the associated domain.

### CODE REVIEW

File: linode/resource_linode_domain_record_test.go [hunk 1]
```diff
@@ -41,6 +41,58 @@ func TestAccLinodeDomainRecord_basic(t *testing.T) {
 	})
 }
 
+func TestAccLinodeDomainRecord_noName(t *testing.T) {
```
Comment (Charliekenney23) [inline linode/resource_linode_domain_record_test.go]: This test validates that non-SRV records cannot be created without a `name`.

File: linode/resource_linode_domain_record_test.go [hunk 2]
```diff
@@ -41,6 +41,58 @@ func TestAccLinodeDomainRecord_basic(t *testing.T) {
...
+	expectedTarget := "mysubdomain." + domainName
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLinodeDomainRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckLinodeDomainRecordConfigSRVNameSet(domainName),
+				ExpectError: regexp.MustCompile(errLinodeDomainRecordSRVNameComputed),
```
Comment (Charliekenney23) [inline linode/resource_linode_domain_record_test.go]: Validates SRV records should not be configured with a name.

File: linode/resource_linode_domain_record_test.go [hunk 3]
```diff
@@ -41,6 +41,58 @@ func TestAccLinodeDomainRecord_basic(t *testing.T) {
...
+			},
+			{
+				Config:      testAccCheckLinodeDomainRecordConfigSRVInvalidTarget(domainName),
+				ExpectError: regexp.MustCompile("Target for SRV records must be the associated domain or a related FQDN."),
```
Comment (Charliekenney23) [inline linode/resource_linode_domain_record_test.go]: Validates SRV record targets must be FQDN/associated domain.

File: linode/resource_linode_domain_record.go [hunk 4]
```diff
@@ -192,18 +199,49 @@ func resourceDataIntOrNil(d *schema.ResourceData, name string) *int {
...
+// validateSRVRecordTarget ensures an SRV domain record's target is a the domain itself or
+// a Fully Qualified Domain Name based on the associated domain.
+func validateSRVRecordTarget(c *linodego.Client, domainID int, target string) error {
+	domain, err := c.GetDomain(context.Background(), domainID)
+	if err != nil {
+		return err
+	}
+
+	if target != domain.Domain && !strings.HasSuffix(target, "."+domain.Domain) {
+		return fmt.Errorf(`Target for SRV records must be the associated domain or a related FQDN. Did you mean "%s.%s"?`, target, domain.Domain)
```
Comment (phillc) [inline linode/resource_linode_domain_record.go]: Nice

File: linode/resource_linode_domain_record.go [hunk 5]
```diff
@@ -192,18 +199,49 @@ func resourceDataIntOrNil(d *schema.ResourceData, name string) *int {
...
 		return fmt.Errorf("Invalid Client when creating Linode DomainRecord")
 	}
 	domainID := d.Get("domain_id").(int)
+	name := d.Get("name").(string)
+	target := d.Get("target").(string)
+
+	typ := linodego.DomainRecordType(d.Get("record_type").(string))
+	isSRVRecord := typ == linodego.RecordTypeSRV
+
+	if name == "" && !isSRVRecord {
```
Comment (phillc) [inline linode/resource_linode_domain_record.go]: that is a lot of if

File: linode/resource_linode_domain_record_test.go [hunk 6]
```diff
@@ -170,3 +222,55 @@ resource "linode_domain_record" "foobar" {
 	target = "target.%s.example"
 }`, domainRecord, domainRecord)
 }
+
+func testAccCheckLinodeDomainRecordConfigNoName(domainName string) string {
```
Comment (phillc) [inline linode/resource_linode_domain_record_test.go]: I like that these map to those conditionals

**Pull request merged**